services/app-api/utils/other/other.ts

```diff
@@ -102,3 +103,63 @@ export const getLastCreatedWorkPlan = async (
   // If there wasn't an eligble work plan to copy from, return undefined
   return { workPlanMetadata: undefined, workPlanFieldData: undefined };
 };
+
+export const getReportYear = (
+  reportType: string,
+  //reportData: unvalidatedMetadata || workPlanMetadata
+  reportData: any,
+  isCopyOver: boolean = false
+): number => {
+  if (isCopyOver) {
+    if (typeof reportData?.copyReport?.reportYear !== "number") {
+      throw new Error("Invalid value for reportYear");
+    }
+    const prevReportYear = reportData?.copyReport?.reportYear;
+    const prevReportPeriod = reportData?.copyReport?.reportPeriod;
+
+    return prevReportPeriod == 2 ? prevReportYear + 1 : prevReportYear;
+  }
+
+  if (
+    (reportType === ReportType.WP && !isCopyOver) ||
+    reportType === ReportType.SAR
+  ) {
+    if (typeof reportData.reportYear !== "number") {
+      throw new Error("Invalid value for reportYear");
+    }
+
+    return reportData?.reportYear;
+  }
+
+  return new Date(convertDateUtcToEt(Date.now())).getFullYear();
```

Comment:
what is the case for this last return? as i read it we must hit one of the former two ifs

Reply:
🤷🏻‍♀️ you are probably right? I kept it as a fallback but we can probably remove it! Will refactor this.